src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -233,14 +243,23 @@ class segment_meta_column_frame
         return sizeof(*this) + _tail.value().mem_use();
     }
 
+    /// Return first element of the frame or nullopt if frame is empty
+    auto get_initial_value() const noexcept {
+        if (_tail.has_value()) {
+            return std::make_optional(_tail->get_initial_value());
+        }
+        return std::nullopt;
+    }
+
     const_iterator begin() const {
         if (_tail.has_value()) {
             decoder_t decoder(
               _tail->get_initial_value(),
               _tail->get_row_count(),
               _tail->share(),
               delta_alg_instance);
-            return const_iterator(std::move(decoder), _head, _size);
+            return const_iterator(
+              std::move(decoder), _head, _size, 0, _tail->get_initial_value());
```

Comment:
nit: it feels unintuitive to be passing the tail to a begin() constructor

Reply:
_tail is the buffer containing the already compressed data of this frame. 

when new data is appended, is recorded into _head. when _head is full (16 elements), its content is compressed into _tail.

in cstore the begin const_interator is not a thin proxy, it contains the whole iobuf of the frame, so that the frame can be deleted and the const_iterator still be valid (not sure if there are any actual use case anymore for this design, but this way it can support concurrent iterations/trucations/substitutions). In fact the end_iterator is mostly a sentinel value